README.md

```diff
@@ -32,3 +32,5 @@ and stop with `make stop-local`.
 
 Please contact a FlowEHR admin to deploy to the FlowEHR infrastructure where a 
 managed repository will be created with credentials to deploy to the cloud.
+
+To add runtime environment variables into your app follow the [example diff](https://github.com/UCLH-Foundry/Dash-Seedling/compare/335ba5c85d8ec9c1e1ecaf32d86fb1c9b96e37a4...15fa63f6f9a62bfece8f35b1ee5c321b3ce22973) to, for example, add a connection string stored as a GitHub secret or variable. This must be performed by a code owner.
```

Comment:
This change is good.

I had a question about the diff itself.  I will add it here (to this PR) but you might want them to live somewhere else...

Why use `ARG` and `ENV` in the Dockerfile?

Generally, you'd use this for build-time env variables that are baked into the image. However, isn't the example you're suggesting a run-time env var? 

Would I know the value of `YOUR_SECRET_CONNECTION_STRING` are docker build time? I'd have thought this would only be known at deploy time.

For example, `FEATURE_STORE_CONNECTION_STRING` is an env var, but not defined in the Docker file in this way.

Reply:
Fair point and perhaps should have been/be explained somewhere.

This is a workaround for not having access to `docker run` to define runtime envrionment variables as that gets executed by the webapp after pulling the image. We wanted a way for developers to be able to set potentially sensitive environment variables without them having to do a FlowEHR PR to update an [apps.yaml](https://github.com/UCLH-Foundry/FlowEHR/blob/5f50dd1693fb38208c3b050e588323c28d493e65/apps/apps.yaml#L12), which is where [app_settings](https://github.com/UCLH-Foundry/FlowEHR/blob/5f50dd1693fb38208c3b050e588323c28d493e65/apps/app/webapp.tf#L48) get set. The `YOUR_SECRET_CONNECTION_STRING` was meant to be an example of a sensitive secret, but not related to any FlowEHR infrastructure

`FEATURE_STORE_CONNECTION_STRING` is set in app_settings as a runtime env var

Reply:
Right...so it's something that is baked into the docker image, and is not environment specific (i.e. dev/test/prod would all have the same value)? Or is there opportunity to set it per environment?

I don't know if I'd want a sensitive secret to be sitting in the image like this. Why work around the PR to apps.yaml? Isn't that the better/more secure way to do this?

Reply:
It is baked into the image but not environment specific, well semi. Unchanged between prod-testing and prod because of the slot swap retaining the image.

Also troubles me a bit, but it’s not being uploaded to docker hub so I don’t feel so bad about it (similar to databricks secrets I guess?). The app_settings are getting stored in plain text anyway, albeit in a soon to be vnet secured container.

PR to FlowEHR might be more secure but harder to manage (FlowEHR environment secrets could get very large) and likely slower time to prod(?)

Reply:
> Even though this is possible, should we be documenting and recommending it? We're pointing users down a dodgy path here.

I'm pretty 50/50. Functionality is useful but agree that we shouldn't be promoting baking secrets into an image.. @jjgriff93 @damoodamoo what're your thoughts?

I'm not sure there's anything we can do to that needs to set runtime vars, because a developer in this repository 
only has credentials to push a docker image (with no testing slot provisioned). I guess another option would be to provide a SP for all environments so you could use [az webapp config appsettings](https://learn.microsoft.com/en-us/cli/azure/webapp/config/appsettings?view=azure-cli-latest) or something similar